packages/vaadin-grid/src/vaadin-grid.js

```diff
@@ -385,18 +385,40 @@ class GridElement extends ElementMixin(
       },
 
       /**
+       * **Deprecated**
+       *
+       * See [`allRowsVisible`](#/elements/vaadin-grid#property-allRowsVisible)
+       *
        * If true, the grid's height is defined by its rows.
        *
        * Effectively, this disables the grid's virtual scrolling so that all the rows are rendered in the DOM at once.
        * If the grid has a large number of items, using the feature is discouraged to avoid performance issues.
+       *
        * @attr {boolean} height-by-rows
        * @type {boolean}
+       * @deprecated since version 14.7
        */
       heightByRows: {
```

Comment:
As discussed in Slack, I would suggest to remove this here, and instead add a getter + setter that uses `allRowsVisible`:
```
get heightByRows() {
  return this.allRowsVisible;
}

set heightByRows(value) {
  this.allRowsVisible = value;
}
```

That would allow us to remove all traces of it in the code, and would make removing it easier later on.

Reply:
we can't change the `heightByRows` property to setter/getters since it's a reflecting property (makes an attribute) and that's part of the public API. Of course we can manually do the work that is necessary (adding/removing the attribute and deserialising to reflect to the property) but it doesn't worth it, I think.

Since it's something that we're going to remove I think not going that path makes more sense.

This state is only referenced in the component once (to add some padding if the agent is Firefox). We can make a method and call that instead of accessing these properties directly.